### ISSUE TITLE
React 16 server renderer breaks with style object in prod mode

### ISSUE THREAD

**Issue by aickin** [facebook/react#10299]
There is a bug in the React 16 beta server renderer, reported by @7rulnik [in the React 16 beta thread](https://github.com/facebook/react/issues/10294#issuecomment-318231160), where the renderer throws an exception when `style` is included and NODE_ENV is `production`.

Given this file:

```
// index.js
const React = require('react')
const ReactServer = require('react-dom/server')
const foo1 = React.createElement("div", {
  className: "sign-link",
  id: "sign-layout-link",
  style: { position: 'absolute' }
});

console.log(ReactServer.renderToString(foo1))
```

If you run `NODE_ENV=production node index.js`, you'll get:

```
TypeError: re is not a function
    at /Users/v7rulnik/projects/kupibilet/internals/kupibilet.ru/node_modules/react-dom/cjs/react-dom-server.production.min.js:1:7774
    at /Users/v7rulnik/projects/kupibilet/internals/kupibilet.ru/node_modules/fbjs/lib/memoizeStringOnly.js:23:32
    at f (/Users/v7rulnik/projects/kupibilet/internals/kupibilet.ru/node_modules/react-dom/cjs/react-dom-server.production.min.js:1:2423)
    at b (/Users/v7rulnik/projects/kupibilet/internals/kupibilet.ru/node_modules/react-dom/cjs/react-dom-server.production.min.js:1:3163)
    at e.renderDOM (/Users/v7rulnik/projects/kupibilet/internals/kupibilet.ru/node_modules/react-dom/cjs/react-dom-server.production.min.js:1:9665)
    at e.render (/Users/v7rulnik/projects/kupibilet/internals/kupibilet.ru/node_modules/react-dom/cjs/react-dom-server.production.min.js:1:8762)
    at e.read (/Users/v7rulnik/projects/kupibilet/internals/kupibilet.ru/node_modules/react-dom/cjs/react-dom-server.production.min.js:1:8452)
    at Object.T [as renderToString] (/Users/v7rulnik/projects/kupibilet/internals/kupibilet.ru/node_modules/react-dom/cjs/react-dom-server.production.min.js:1:4384)
    at repl:1:13
    at ContextifyScript.Script.runInThisContext (vm.js:44:33)
```

Weirdly, dev mode does **not** throw an error.

**Comment by gzhappysky** [comment]
Because `ReactDOMInvalidARIAHook` has a [getStackAddendumByID](https://github.com/facebook/react/blob/master/src/renderers/dom/shared/hooks/ReactDOMInvalidARIAHook.js#L25) variable that be declared by DEV env. But [It](https://github.com/facebook/react/blob/master/src/renderers/dom/shared/hooks/ReactDOMInvalidARIAHook.js#L31) can used by production env.

**Comment by gaearon** [comment]
Fixed in `16.0.0-beta.2`.
